internal/controller/apiextensions/composite/composition_pt.go

```diff
@@ -273,6 +273,16 @@ func (c *PTComposer) Compose(ctx context.Context, xr *composite.Unstructured, re
 		o := []resource.ApplyOption{resource.MustBeControllableBy(xr.GetUID()), usage.RespectOwnerRefs()}
 		o = append(o, mergeOptions(filterPatches(t.Patches, patchTypesFromXR()...))...)
 		if err := c.client.Apply(ctx, cd, o...); err != nil {
+			if kerrors.IsInvalid(err) {
+				// NOTE(phisco): we used to catch invalid resources as a
+				// side-effect of the dry-run apply we run to generate a name
+				// for them. We no longer do that, so here we'll just ignore
+				// these errors and remove the resource from cds so that it's
+				// not going to be observed later.
+				cds[i] = nil
```

Comment:
we also want to emit an event here actually, I was doublechecking whether we did it on 1.13 and we did: https://github.com/crossplane/crossplane/blob/01c5d2b0308e775e2043f9552631037c23ceaed8/internal/controller/apiextensions/composite/composition_pt.go#L212

Reply:
Do we need a similar change in composition_functions.go - https://github.com/crossplane/crossplane/blob/13c2d1f42c61d332b065f3b20c1db6e10c773c37/internal/controller/apiextensions/composite/composition_functions.go#L409 ?

Reply:
Yes, I have to test it, but from the code I'd say so, the code was based on the same assumption

Reply:
looks like we always returned early in case of errors when dry-running resources with functions 🤔 https://github.com/crossplane/crossplane/pull/4858/files#diff-e62ddfeee06c113f3ce89d4236b1a8295f2c6e0158d3abc07a866dfb36c75e63L326